File: mobile/lib/models/service.dart
class Category {
  final String id;
  final String name;
  final String? icon;
  final String colorHex;
  final bool isActive;

  Category({
    required this.id,
    required this.name,
    this.icon,
    required this.colorHex,
    required this.isActive,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'],
      name: json['name'],
      icon: json['icon'],
      colorHex: json['color_hex'],
      isActive: json['is_active'],
    );
  }
}

class Service {
  final String id;
  final String categoryId;
  final String name;
  final String? description;
  final double minPrice;
  final double maxPrice;
  final int urgencyLevel;
  final bool isActive;

  Service({
    required this.id,
    required this.categoryId,
    required this.name,
    this.description,
    required this.minPrice,
    required this.maxPrice,
    required this.urgencyLevel, 
    required this.isActive,
  });

  factory Service.fromJson(Map<String, dynamic> json) {
    return Service(
      id: json['id'],
      categoryId: json['category_id'],
      name: json['name'],
      description: json['description'],
      minPrice: (json['min_price'] ?? 0).toDouble(),
      maxPrice: (json['max_price'] ?? 0).toDouble(),
      urgencyLevel: json['urgency_level'] ?? 1,
      isActive: json['is_active'] ?? true,
    );
  }
}

File: backend/app/models/service.py
from sqlalchemy import Column, String, Float, Boolean, Integer, ForeignKey, Text
from sqlalchemy.orm import relationship
from app.core.database import Base
import uuid

def gen_uuid(): return str(uuid.uuid4())

class Category(Base):
    __tablename__ = "categories"
    id = Column(String, primary_key=True, default=gen_uuid)
    name = Column(String(100), nullable=False)
    icon = Column(String(100))
    color_hex = Column(String(7), default="#000000")
    is_active = Column(Boolean, default=True)
    services = relationship("Service", back_populates="category")

class Service(Base):
    __tablename__ = "services"
    id = Column(String, primary_key=True, default=gen_uuid)
    category_id = Column(String, ForeignKey("categories.id"))
    name = Column(String(150), nullable=False)
    description = Column(Text)
    min_price = Column(Float, default=0)
    max_price = Column(Float, default=0)
    urgency_level = Column(Integer, default=1)
    is_active = Column(Boolean, default=True)
    category = relationship("Category", back_populates="services")
